src/server/project.ts

```diff
@@ -81,6 +86,14 @@ namespace ts.server {
             return this.languageService;
         }
 
+        getCompileOnSaveAffectedFileList(triggerFileName: string): string[] {
+            if (!this.languageServiceEnabled) {
+                return [];
+            }
+            this.updateGraph();
+            return this.builder.getFilesAffectedBy(triggerFileName);

```

Comment:
I think you can 
- change the signature of `getFilesAffectedBy` to accept `Path`
- get a `ScriptInfo` for a `triggerFileName` and then pass `ScriptInfo.path` to `getFilesAffectedBy`

This will allow to avoid conversion `string -> Path` in several places